go/vt/vttablet/tabletmanager/vreplication/engine.go

```diff
@@ -844,6 +856,22 @@ func (vre *Engine) readAllRows(ctx context.Context) ([]map[string]string, error)
 	return maps, nil
 }
 
+func (vre *Engine) getAutoIncrementStep(dbClient binlogplayer.DBClient) (int32, error) {
+	qr, err := dbClient.ExecuteFetch("select @@auto_increment_increment", 1)
+	if err != nil {
+		return 0, err
+	}
+	if len(qr.Rows) != 1 {
+		// Handles case where underlying mysql doesn't support auto_increment_increment for any reason.
+		return 1, nil
+	}
+	autoIncrement, err := evalengine.ToInt64(qr.Rows[0][0])
```

Comment:
In light of all the recent CodeQL related work and aligning types, we should match what MySQL uses: https://dev.mysql.com/doc/refman/8.0/en/replication-options-source.html#sysvar_auto_increment_increment

MySQL uses a uint16 for this variable. Doesn't look like we've yet added `ToUint16()`, so we could do that here: https://github.com/vitessio/vitess/blob/main/go/sqltypes/value.go#L319
```
func (v Value) ToUint16() (uint16, error) {
	if !v.IsIntegral() {
		return 0, ErrIncompatibleTypeCast
	}

	u, err := strconv.ParseUint(v.RawStr(), 10, 16)
	return uint16(u), err
}
```

This will @dbussink and I some future work. 😄

Reply:
Yeah, this would be nice to keep things correctly typed and so we throw errors that are correct when there's a type mismatch instead of more silent weird truncation.

Reply:
Fixed